File: Backend/models/Notice/NoticeModel.js
const mongoose = require('mongoose');
const moment = require("moment");
const userB = require('../UserBasicModel');
const userBasicSchema = require('../UserBasicModel');
const DiscussionModel = require('../DiscussionModel');
const Concent = require('./ConcentModel');

const NoticeModel = new mongoose.Schema({
    title : String,
    description : String,
    docLink : String,
    timeStamp : String,
    AcknowledgeBy : [userB.schema],
    Discussions : [DiscussionModel.schema],
    Concents : [{type : mongoose.Schema.Types.ObjectId, ref : "Concent"}],
}, {collection : 'Notices'});

NoticeModel.methods.addDiscussion = async function(userBID, payload, messageType) {
    const userInfo = await userBasicSchema.findById(userBID);
    const timeStamp = moment().format();
    const discussion = new DiscussionModel({userInfo, payload, messageType, timeStamp});
    this.Discussions.push(discussion);
    discussion.save();
}

NoticeModel.methods.addAcknowledgement = async function(userBID){
    const userInfo = await userBasicSchema.findById(userBID);
    this.AcknowledgeBy.push(userInfo);
}

module.exports = mongoose.model('NoticeModel', NoticeModel);

File: frontend/lib/Classes/NoticeModel.dart
import 'package:frontend/Classes/DiscussionModel.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:moment/moment.dart';
import 'UserB.dart';
part 'NoticeModel.g.dart';

@JsonSerializable()
class NoticeModel {
  NoticeModel(
      {required this.id,
      required this.title,
      required this.description,
      required this.docLink,
      required this.timeStamp,
      required this.AcknowledgeBy,
      required this.Discussions});

  factory NoticeModel.fromJson(Map<String, dynamic> data) =>
      _$NoticeModelFromJson(data);
  Map<String, dynamic> toJson() => _$NoticeModelToJson(this);

  final String id;
  final String title;
  final String description;
  final String docLink;
  final DateTime timeStamp;
  final List<UserB> AcknowledgeBy;
  final List<DiscussionModel> Discussions;
}
